### PR TITLE
Added myself to members, created members for hire section, added link to...

### PR DESCRIPTION
**Pull request by letyco**
Add a members for hire section to the site.

This pull request creates a members for hire section, a link to the members for hire section in the sidebar, and a list of members (in the data directory) to show there.

The pull request also includes a single addition to the members section.

### CODE REVIEW

File: source/members-for-hire/index.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+created_by:
+  email: james@lety.co
+---
+
+A list of LRUG members who are available to hire on a contract basis. These individuals usually attend LRUG [meetings](/meetings/), or have attended a few of them, or maybe even just one.  Maybe they just signed up to the mailing-list, whatever, it's not an exclusive club or anything.
+
+<ul>
+  <% data.hire.each do |member| %>
+    <li class="vcard">
+      <a class="url" href="<%= member.url %>" title="<%= member.url %>">
```
Comment (andyw8) [inline source/members-for-hire/index.html.md.erb]: Is there a reason for setting the `title` as a URL?

**Comment by andyw8** [comment]
In the LRUG thread, you wrote:

> I am unaware of any apps out there that already do this, which we could either link to or pull status updates/availability via Javascript or JSONP or whatever.

GitHub has a simple "Available for Hire" boolean field, so if there was a way for members to link their GitHub account to the LRUG site then that could work.

File: data/hire.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+// https://github.com/lrug/lrug.org/pulls
+// Do keep the list sorted by last name when you do so.
+// We're using microformats as the keys here, although we currently only
+// render out fn and url. Feel free to add more keys and we'll update the
+// renderer.
+[
+  {
+    "fn":"James R. McCarthy, LetyCo",
+    "url":"http://lety.co/"
+  }
```
Comment (andyw8) [inline data/hire.json]: How about some way of indicating whether recruiters may contact you?

**Comment by letyco** [comment]
> GitHub has a simple "Available for Hire" boolean field, so if there was a way for members to link their GitHub account to the LRUG site then that could work.

From what I understand of the github documentation, this boolean only allows you to apply for jobs that are listed on github (showing "appropriate" ones to you), and does not allow you to use this as a method of advertising your availability on your profile.

File: data/hire.json (hunk 2 above)
Comment (letyco) [inline data/hire.json]: Surely if the user feels strongly about this they can indicate so on the site they link to. I fail to see a good reason to include more than a link to a site of your choice.


At first I thought that indicating whether you we're presently available was a good idea, but then realised that this info would go stale without constant updating on the users part, which would bring in the requirement for there to be an app or system allowing users to easily keep their status updated.

Given the small numbers involved in this it's not really worthwhile.

**Comment by andyw8** [comment]
Hmm, I can't see any restrictions like that in the API documentation, it seems to be part of the public /users/ API: https://developer.github.com/v3/users/

**Comment by letyco** [comment]
My bad, was looking at user documentation, not API docs (also been a long day for me).

We could add a github account field (along with website and name) and inform people who have asked to be included on the list if they want to make use of this.

A bit of JQuery would be able to fetch the details (email address, website, availability)

**Comment by andyw8** [comment]
Yeah, that sounds feasible. If it's done client-side with no way of caching the result then then you *might* run into problems hitting the rate limit though, I'm not sure how generous it is.

**Comment by letyco** [comment]
Rate limit is not an issue until this list starts to grow quite large (with one API call per user)

**Comment by h-lame** [comment]
@letyco FYI: I've taken the "add myself to members" bit out of this commit and rolled it into master directly (see a7ab8d748cbd8607a96203fe81b9fb86add37171)

File: source/members-for-hire/index.html.md.erb (hunk 1 above)
Comment (letyco) [inline source/members-for-hire/index.html.md.erb]: I believe this is dictated by the directory name, currently the directory is named members-for-hire and that is the url structure.

**Comment by letyco** [comment]
andyw8 I'm thinking that live availability via github accounts should be a separate pull request, it is additional functionality over the addition of a members for hire list (which would be the default view without Javascript, which LRUG.org currently doesn't require).

I still think it should be done (and once this pull request is approved/rejected will begin working on it).

**Comment by h-lame** [comment]
I was chatting to @andrewmcdonough and he suggested something a bit easier to manage using google spreadsheets or something.  Good to see a bunch of names on the list though, certainly suggests there's some appetite for it.  I'll hold off on merging though until Andrew chips in and we see if his approach is something we want to go with.

**Comment by letyco** [comment]
No probs,

talk later.

James.
On 23/02/15 14:46, Murray Steele wrote:
>
> I was chatting to @andrewmcdonough 
> <https://github.com/andrewmcdonough> and he suggested something a bit 
> easier to manage using google spreadsheets or something. Good to see a 
> bunch of names on the list though, certainly suggests there's some 
> appetite for it. I'll hold off on merging though until Andrew chips in 
> and we see if his approach is something we want to go with.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/lrug/lrug.org/pull/30#issuecomment-75553044>.
>

-- 
James McCarthy

Software Consultant

LetyCo

Mob:  07577006897

Email:  james@lety.co

lety.co

**Comment by h-lame** [comment]
This is embarrassingly slow, but I'm going to close this.  See discussion on #31, but basically, we're less and less convinced of the benefits of something croud-sourced like this or the existing `members-and-friends` page because after an initial flurry of excitement it'll probably just rot.  There are probably better and more accurate places to find out who is / isn't looking for work (LinkedIn, hired, etc...).

Sorry for leaving it soooo long without any closure either way.  I suspect even if we had merged it, based on the discussion in #31 we'd probably be deleting it soon along with `members-and-friends`.  Thanks for trying something though!

**Pull request closed without merging**